rcl/test/rcl/test_init.cpp

```diff
@@ -271,3 +280,24 @@ TEST_F(CLASSNAME(TestRCLFixture, RMW_IMPLEMENTATION), test_rcl_get_instance_id)
   ret = rcl_context_fini(&context);
   EXPECT_EQ(ret, RCL_RET_OK);
 }
+
+TEST_F(CLASSNAME(TestRCLFixture, RMW_IMPLEMENTATION), test_rcl_init_options_access) {
+  rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+  rcl_ret_t ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+
+  rmw_init_options_t * options = rcl_init_options_get_rmw_init_options(&init_options);
+  ASSERT_NE(nullptr, options);
+  EXPECT_EQ(0u, options->instance_id);
+  EXPECT_EQ(nullptr, options->impl);
+
+  const rcl_allocator_t * options_allocator = rcl_init_options_get_allocator(&init_options);
+  EXPECT_TRUE(rcutils_allocator_is_valid(options_allocator));
+
+  rcl_init_options_t init_options_dst = rcl_get_zero_initialized_init_options();
+  EXPECT_EQ(RCL_RET_OK, rcl_init_options_copy(&init_options, &init_options_dst));
+  EXPECT_EQ(RCL_RET_ALREADY_INIT, rcl_init_options_copy(&init_options, &init_options_dst));
+
+  EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options));
```

Comment:
@Blast545 what if the assertion in [line 290](https://github.com/ros2/rcl/pull/657/files#diff-c67929cc01d6cd07de8ab134e5d2cfd2R290) fails?

Reply:
AFAIK, the test will not continue if that `ASSERT_NE` fails. Am I missing something from your question?

Reply:
Correct, in which case this `rcl_init_options_fini()` call won't be executed.

Reply:
Oh, thanks! Totally missed that. Fixed in the last commit.